apps/web/src/components/HoverCard/HoverCardCommunityInner.tsx

```diff
@@ -138,16 +138,25 @@ export function HoverCardCommunityInner({
     return null;
   }
 
+  const formatEthAddress = (e: string) => {
+    return e.slice(0, 4) + '...' + e.slice(-4);
+  };
+
+  // check if name is eth address and truncate
+  const hasAddress = Boolean(community.name.startsWith('0x') && community.name.length > 20);
```

Comment:
hmm how did we decide on `20` to be the arbitrary cutoff for name length? i think a more rigorous check would be something like this:

```ts
function isValidEthereumAddress(address: string): boolean {
  const pattern = /^0x[a-fA-F0-9]{40}$/;
  return pattern.test(address);
}
```

^ let's define this function in a new file, like `shared/utils/isValidEthereumAddress`, then use it in this component:


```tsx
import { isValidEthereumAddress } from '~/shared/utils/isValidEthereumAddress'
import { truncateAddress } from '~/shared/utils/wallet';

...

<StyledCardTitle>
  {isValidEthereumAddress(community.name) ? truncateAddress(community.name) : community.name}
</StyledCardTitle>
```

Reply:
moving this function inside wallet utils, since it checks thats its a valid eth wallet address.